librad/src/net/peer.rs

```diff
@@ -380,40 +410,51 @@ where
     S: Signer + Clone,
     S::Error: keys::SignError,
 {
-    fn git_fetch<'a>(
+    async fn git_fetch<'a>(
         &'a self,
         from: &PeerId,
         urn: Either<RadUrn, Originates<RadUrn>>,
         head: impl Into<Option<git2::Oid>>,
-    ) -> Result<(), GitFetchError> {
-        let git = self.inner.lock().unwrap();
+    ) -> Result<(), PeerStorageError> {
+        let git = self.inner.get().await?;
         let urn = urn_context(git.peer_id(), urn);
-
-        if let Some(head) = head.into() {
-            if git.has_commit(&urn, head)? {
-                return Err(GitFetchError::KnownObject(head));
+        let head = head.into();
+        let from = from.clone();
+
+        spawn_blocking(move || {
+            if let Some(head) = head {
+                if git.has_commit(&urn, head)? {
+                    return Err(PeerStorageError::KnownObject(head));
+                }
             }
-        }
 
-        let url = RadUrl {
-            authority: from.clone(),
-            urn,
-        };
-        git.fetch_repo(url, None).map_err(|e| e.into())
+            let url = RadUrl {
+                authority: from,
+                urn,
+            };
+            git.fetch_repo(url, None).map_err(|e| e.into())
+        })
+        .await
+        .expect("`PeerStorage::git_fetch` panicked")
     }
 
     /// Determine if we have the given object locally
-    fn git_has(
+    async fn git_has(
         &self,
         urn: Either<RadUrn, Originates<RadUrn>>,
         head: impl Into<Option<git2::Oid>>,
     ) -> bool {
-        let git = self.inner.lock().unwrap();
+        let git = self.inner.get().await.unwrap();
         let urn = urn_context(git.peer_id(), urn);
-        match head.into() {
+        block_in_place(|| match head.into() {
             None => git.has_urn(&urn).unwrap_or(false),
             Some(head) => git.has_commit(&urn, head).unwrap_or(false),
-        }
+        })
+    }
+
+    async fn is_tracked(&self, urn: &RadUrn, peer: &PeerId) -> Result<bool, PeerStorageError> {
+        let git = self.inner.get().await?;
+        Ok(block_in_place(|| git.is_tracked(urn, peer))?)
```

Comment:
When we call `block_in_place` here, does that mean we force the threaded runtime on consumers as alluded to in a FIXME further up?